inc/integrations/namespace.php

```diff
@@ -72,7 +72,14 @@ function auto_register_plugin_integrations() {
  */
 function load_integrations( array $integrations ) {
 	foreach ( $integrations as $type => $class ) {
-		$integration = new $class();
-		$integration->setup();
+
+		// Ensure the integration exists, and has an `instance` method using a
+		// singleton pattern.
+		if ( ! class_exists( $class ) || ! method_exists( $class, 'instance' ) ) {
+			continue;
+		}
+
+		// Create a singleton instance of this integration.
+		$integration = call_user_func( [ $class, 'instance' ] );
```

Comment:
This could be simplified a bit.

```suggestion
		// Ensure the integration exists and has an instance method.
		if ( is_callable( [ $class, 'instance' ] ) ) {
			// Create a singleton instance of this integration.
			$integration = call_user_func( [ $class, 'instance' ] );
		}

```